jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -486,9 +486,13 @@ static int getServerPort(Request request)
             return authority.getPort();
 
         // Is there a scheme with a default port?
-        HttpScheme scheme = HttpScheme.CACHE.get(request.getHttpURI().getScheme());
-        if (scheme != null && scheme.getDefaultPort() > 0)
-            return scheme.getDefaultPort();
+        String rawScheme = uri.getScheme();
+        if (StringUtil.isNotBlank(rawScheme))
+        {
+            HttpScheme scheme = HttpScheme.CACHE.get(rawScheme);
+            if (scheme != null && scheme.getDefaultPort() > 0)
+                return scheme.getDefaultPort();
+        }
```

Comment:
I think most (all?) of our `Index.get(String, int, int)` are robust to having a null passed.  Some even support a null key value I think.

So I think it might be better to fix the `AbstractIndex` impl of `Index.get` to:
```
    public V get(String s)
    {
        return get(s, 0, s == null ? 0 : s.length());
    }
```

@lorban thoughts?

Reply:
I second that. We can easily consider a null key to be equivalent to a zero-length string when null keys are not supported so I see no reason why we shouldn't do that. That should be reflected in the `Index` javadoc too.